cache/grpcproxy/grpcproxy_test.go

```diff
@@ -0,0 +1,203 @@
+package grpcproxy
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"io"
+	"math/rand"
+	"testing"
+	"time"
+
+	"github.com/buchgr/bazel-remote/v2/cache"
+	"github.com/buchgr/bazel-remote/v2/cache/disk"
+	"github.com/buchgr/bazel-remote/v2/server"
+	"github.com/google/uuid"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"google.golang.org/protobuf/proto"
+
+	pb "github.com/buchgr/bazel-remote/v2/genproto/build/bazel/remote/execution/v2"
+	testutils "github.com/buchgr/bazel-remote/v2/utils"
+	bs "google.golang.org/genproto/googleapis/bytestream"
+)
+
+var logger = testutils.NewSilentLogger()
+
+type fixture struct {
+	port    int
+	host    string
+	cc      *grpc.ClientConn
+	server  *grpc.Server
+	clients *GrpcClients
+	cache   disk.Cache
+}
+
+func newFixture(t *testing.T, proxy cache.Proxy) *fixture {
+	port := rand.Intn(65353-1024) + 1024
+	host := fmt.Sprintf("localhost:%d", port)
+	cc, err := grpc.Dial(host, grpc.WithTransportCredentials(insecure.NewCredentials()))
```

Comment:
Could we use google.golang.org/grpc/test/bufconn for this instead? See the example in server/grpc_test.go